jax/_src/numpy/linalg.py

```diff
@@ -469,12 +470,19 @@ def norm(x, ord=None, axis : Union[None, Tuple[int, ...], int] = None,
       # code has slightly different type promotion semantics, so we need a
       # special case too.
       return jnp.sum(jnp.abs(x), axis=axis, keepdims=keepdims)
-    else:
+    elif isinstance(ord, numbers.Number):
```

Comment:
Technically the $p$-norm is only defined if $p > 1$. To match numpy, I allow $p \leq 1$. Should we diverge from numpy and add a check that it's a valid norm? The math still works for undefined $p$ values so there aren't any numerical issues.